tutorials/11-probabilistic-pca/11_probabilistic-pca.jmd

```diff
@@ -285,31 +250,38 @@ df_rec[!, :cell] = 1:n_cells
 )
 ```
 
-We observe that, the recovered (posterior mean predicted) $\hat{x}$ values align with the original $x$ values - particularly the same pattern in the first and last 3 geme features are captured, which implies the inference is successful.
-This is satisfying as we have just projected the original 9-dimensional space onto a 2-dimensional space - some info has been cut off in the projection process, but we haven't lost any important info, e.g. the key difference between the two groups.
-The is the desirable property of PCA: it picks up the principal axes along which most of the (original) data data variations cluster, and remove those less relevant.
-If we choose the reduced space dimension $k$ to be exactly $D$, we would recover exactly the same input matrix $x$, as the new (projected) space contains the same information as the input space as long as we keep all the dimensions.
-
-Now we have represented the original high-dimensional data in two dimensions, without lossing the key information about the two groups of cells in the input data.
-Finally, the benefits of performing PCA is to analyse and visualise the dimension-reduced data in the projected, low-dimensional space.
+```julia
+heatmap(
+    mat_rec;
+    Colors=:value,
+    xlabel="cell number",
+    yflip=true,
+    ylabel="gene feature",
+    yticks=1:9,
+    colorbar_title="expression",
+)
+```
 
-We can also check the average magnitudes of the gap between $\hat{x}$ and $x$:
+We can quantitatively check the absolute magnitudes of the column average of the gap between `mat_exp` and `mat_rec`:
 
-```julia; echo=false
-let
-    diff_matrix = X' - expression_matrix
-    @assert abs(mean(diff_matrix[:, 4])) < 0.5
-    @assert abs(mean(diff_matrix[:, 5])) < 0.5
-    @assert abs(mean(diff_matrix[:, 6])) < 0.5
-end
+```julia
+diff_matrix = mat_exp .- mat_rec
+abs(mean(diff_matrix[:, 4]))
+abs(mean(diff_matrix[:, 5]))
+abs(mean(diff_matrix[:, 6]))
 ```
```

Comment:
Why was this changed?
The `@assert` are important to easily check whether the values are small enough. And `echo=false` avoids printing the final `true`/`false`.

Reply:
Many thanks for letting me know - just recovered them.